Prevent clicking on flags from revealing

noloerino opened the pull request:
When the user clicks on a flagged square, tiles are no longer revealed. If a flagged square is a zero and would get revealed by an adjacent zero being revealed (but not by a user clicking on it), the flag is still overwritten.

nicolaschan commented:
Could we just change the [client side reveal logic](https://github.com/nicolaschan/mines/blob/a56188d440551a89fc1a553c8f68d80a386f6a43/js/client.js#L70-L87)? 

Or is there some reason (e.g., consistency) that the server needs to keep track of this?

noloerino commented:
I earlier tried adding an extra check on `if (board[parseInt($(this).prop('id').split('-')[0])][parseInt($(this).prop('id').split('-')[1])].flagged`, but that field seems not to be updated properly.

nicolaschan commented:
Ah I see. This is just the initial board that is sent. It doesn't look like the actual state is kept anywhere in except in the CSS and [text property of the square](https://github.com/nicolaschan/mines/blob/a56188d440551a89fc1a553c8f68d80a386f6a43/js/client.js#L165). I guess a sketchy way to do this would be to inspect the HTML element of the square and see if that is a flag.
